app/javascript/elm/src/Popup.elm

```diff
@@ -16,38 +18,38 @@ type PopupPart
     | OtherPart
 
 
-view : msg -> (String -> msg) -> Bool -> Popup -> Html msg
-view toggle onSelect isPopupExtended popup =
-    case popup of
-        SelectFrom ( main, others ) itemType selectedItem ->
-            case ( List.isEmpty main, List.isEmpty others ) of
-                ( True, _ ) ->
-                    div [ id "popup", class "filter-popup" ]
-                        [ selectableItems MainPart others onSelect selectedItem ]
-
-                ( False, True ) ->
-                    div [ id "popup", class "filter-popup" ]
-                        [ selectableItems MainPart main onSelect selectedItem
-                        ]
+viewListPopup : msg -> (String -> msg) -> Bool -> ( List String, List String ) -> String -> String -> Html msg
+viewListPopup toggle onSelect isListExpanded ( main, others ) itemType selectedItem =
+    case ( List.isEmpty main, List.isEmpty others ) of
+        ( True, _ ) ->
+            div [ id "popup", class "filter-popup" ]
+                [ selectableItems MainPart others onSelect selectedItem ]
 
-                ( False, False ) ->
-                    div [ id "popup", class "filter-popup" ]
-                        [ selectableItems MainPart main onSelect selectedItem
-                        , if List.isEmpty others then
-                            text ""
+        ( False, True ) ->
+            div [ id "popup", class "filter-popup" ]
+                [ selectableItems MainPart main onSelect selectedItem
+                ]
 
-                          else if isPopupExtended then
-                            div [ class "filter-popup__more" ]
-                                [ selectableItems OtherPart others onSelect selectedItem
-                                , togglePopupStateButton ("fewer " ++ itemType) toggle
-                                ]
+        ( False, False ) ->
+            div [ id "popup", class "filter-popup" ]
+                [ selectableItems MainPart main onSelect selectedItem
+                , if List.isEmpty others then
+                    text ""
 
-                          else
-                            togglePopupStateButton ("more " ++ itemType) toggle
+                  else if isListExpanded then
+                    div [ class "filter-popup__more" ]
+                        [ selectableItems OtherPart others onSelect selectedItem
+                        , togglePopupStateButton ("fewer " ++ itemType) toggle
                         ]
 
-        None ->
-            text ""
+                  else
+                    togglePopupStateButton ("more " ++ itemType) toggle
+                ]
+
+
+viewEmailForm : Html msg -> Html msg
+viewEmailForm emailForm =
+    emailForm
```

Comment:
I wonder if we need this function, cause it doesn't do anything?

Reply:
😅